app/frontend/src/cljs/teet/file/file_view.cljs

```diff
@@ -294,7 +295,9 @@
                               [buttons/button-primary
                                {:id (str "confirmation-confirm")
                                 :disabled @progress?
-                                :on-click #(do (e! (file-controller/->UploadNewVersion file @selected-file))
+                                :on-click #(do (e! (file-controller/->UploadNewVersion file
+                                                                                       @selected-file
+                                                                                       (:db/id task)))
```

Comment:
The component gets `task` as argument, so I figured it's simpler to pass it to the event rather than dig it from the app state in the event itself.